src/operator/subgraph/build_subgraph.cc

```diff
@@ -450,17 +450,18 @@ void FindInputEntries(const nnvm::Graph& g,
  * \param simple_nods vector of simple nodes in top sorted order
  * \param subgraph_nodes vector of pointers of simples of a subgraph.
  * \param entry_top_order_map mapping entry pointer to its top sorted position
- * \param output_entries output entries of the subgraph
+ * \param output_map output entries of the subgraph
  */
 void FindOutputEntries(nnvm::Graph* g,
                        const std::vector<BiDirectedNodePtr>& simple_nodes,
                        const std::vector<BiDirectedNode*>& subgraph_nodes,
                        const std::unordered_map<const nnvm::NodeEntry*, size_t>&
                          entry_top_order_map,
-                       std::vector<nnvm::NodeEntry*>* output_entries) {
+                       std::map<int, std::vector<nnvm::NodeEntry*> > &output_map) {
```

Comment:
Use `std::vector<std::vector<nnvm::NodeEntry*>>` instead. This is fast(O(1) for search and push_back), and can avoid element missing(for std::map, we need additional check to ensure that for each i < output_map.size(), output_map[i] exists).